campaign-launcher/client/src/providers/NetworkProvider.tsx

```diff
@@ -24,9 +25,21 @@ const NetworkProviderContext = createContext<
 export const NetworkProvider: FC<PropsWithChildren> = ({ children }) => {
   const config = useConfig();
   const [appChainId, setAppChainId] = useState<ChainId>(config.chains[0].id);
-  const [isSwitching, setIsSwitching] = useState(false);
+  const [isSwitching, setIsSwitching] = useState(true);
   const { switchChainAsync } = useSwitchChain();
 
+  useEffect(() => {
+    const initChainId = async () => {
+      const savedChainId = await config?.storage?.getItem('chainId') as ChainId | null;
```

Comment:
now, if you're changing the app's network and then reload the page, the app network would always be polygon (amoy). let's persist a selected network and use it after page reloads